project-koku/koku
# COST-347: Renaming aws org unit results in ou being counted twice

Jira Issue: https://issues.redhat.com/browse/COST-347

#### Bug Breakdown:
Whenever an organizational unit is renamed the original name and the new name are both showing up in the cost report endpoint here: 

http://127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?group_by%5Borg_unit_id%5D=R_001

The best way to replicate this issue is to rename the following org unit here: https://github.com/project-koku/koku/blob/master/scripts/aws_org_tree.yml#L88

and run the make target `make load-aws-org-unit-tree` to create the tree.

You will see that two dictionaries are returned:
```
{
                    "id": "OU_001",
                    "type": "organizational_unit",
                    "date": "2020-07-05",
                    "values": [
                        {
                            "date": "2020-07-05",
                            "source_uuid": [
                                "2cba4190-c4f8-43dd-8669-cf55cbc6d8c5"
                            ],
                            "infrastructure":{...},
                            "supplementary": {...},
                            "cost": {...},
                            "id": "OU_001",
                            "alias": "Dept OU_001"
                        }
                    ]
                },
                {
                    "id": "OU_001",
                    "type": "organizational_unit",
                    "date": "2020-07-05",
                    "values": [
                        {
                            "date": "2020-07-05",
                            "source_uuid": [
                                "2cba4190-c4f8-43dd-8669-cf55cbc6d8c5"
                            ],
                            "infrastructure": {...},
                            "supplementary": {...},
                            "cost": {...},
                            "id": "OU_001",
                            "alias": "Dept OU_001 (Rename)"
                        }
```
You will notice that there is one dictionary for the old alias `Dept OU_001` and one for the new alias `Dept OU_001 (Rename)`

---- 

#### How to test

I have already added the scenario of the org unit being renamed to our `aws_org_tree.yml` file. Therefore, we just need to run the make target to load the tree with `make load-aws-org-unit-tree` and then check our cost endpoint to make sure only one dictionary shows up **per day** for `OU_001`. Also, double check to make sure the old alias doesn't show up anywhere.
